compiler/src/test/scala/org/scalajs/core/compiler/test/JSUndefinedParamTest.scala

```diff
@@ -0,0 +1,91 @@
+package org.scalajs.core.compiler.test
+
+import language.experimental.macros
+
+import org.scalajs.core.compiler.test.util._
+import org.junit.Test
+
+import java.net.URLClassLoader
+
+/** Dummy object to get the right shadowing for 2.10 / 2.11 cross compilation */
+private object Compat210 {
+  object blackbox { // scalastyle:ignore
+    type Context = scala.reflect.macros.Context
+  }
+}
+
+import Compat210._
+
+/** This tests the UndefinedParam tracker in the compiler backend.
+ *
+ *  In order to properly implement removal of trailing default parameters, the
+ *  compiler backend may generate a UndefinedParam tree and catch it later.
+ *  However, some macros and compiler plugins may generate trees the backend
+ *  doesn't expect. As a result the backend used to generate invalid IR.
+ *  Instead, we now track these helper trees and emit a more helpful error
+ *  message if one of them sticks around.
+ *
+ *  This test contains a macro that generates a tree we cannot handle and
+ *  verifies that the compiler bails out.
+ */
+object JSUndefinedParamTest {
+  import scala.reflect.macros._ // shadows blackbox from above
+  import blackbox.Context
+
+  def extractArg(call: Any): Any = macro extractArg_impl
+
+  def extractArg_impl(c: Context)(call: c.Expr[Any]): c.Expr[Any] = {
+    import c.universe._
+
+    call.tree match {
+      case Apply(fun, List(arg)) => c.Expr[Any](arg)
+
+      case tree =>
+        c.abort(tree.pos, "Bad tree. Need function call with single argument.")
+    }
+  }
+}
+
+// scalastyle:off line.size.limit
+
+class JSUndefinedParamTest extends DirectTest with TestHelpers {
+
+  override def classpath: List[String] = {
+    // Make JSUndefinedParamTest available to the code we compile in the test
+    val loader =
+      JSUndefinedParamTest.getClass.getClassLoader.asInstanceOf[URLClassLoader]
```

Comment:
This is not scalac's `DirectTest`. It's our own. So, no, it is not able to do that. But we can implement it if we want :)